test/sqllogictest/type-promotion.slt

```diff
@@ -280,7 +280,7 @@ EXPLAIN RAW PLAN FOR
 ----
 %0 =
 | Constant ()
-| Map lower(chartostr(strtochar("a")))
+| Map lower(chartostr(strtochar(strtochar("a"))))
```

Comment:
The doubled call to `strtochar` are a little mind-bending to me--do you have immediate intuition why the intermediate call's getting injected where it wasn't before? Glad to look at this if the answer doesn't come to you in like 10 seconds.